Source/WebCore/platform/mac/LocalCurrentGraphicsContextMac.mm

```diff
@@ -26,33 +26,25 @@
 
 namespace WebCore {
 
-LocalCurrentGraphicsContext::LocalCurrentGraphicsContext(GraphicsContext& graphicsContext, bool isFlipped)
-    : m_savedGraphicsContext(graphicsContext)
+LocalCurrentContextSaver::LocalCurrentContextSaver(CGContextRef cgContext, bool isFlipped)
 {
-    m_savedGraphicsContext.save();
-
-    if (!m_savedGraphicsContext.hasPlatformContext()) {
-        WTFLogAlways("LocalCurrentGraphicsContext is setting the global context to nil because the provided GraphicsContext does not have a platform context (likely display list recording)");
+    if (!cgContext) {
+        ASSERT_NOT_REACHED();
         [NSGraphicsContext setCurrentContext:nil];
         return;
     }
-
-    CGContextRef cgContext = this->cgContext();
     if (cgContext == [[NSGraphicsContext currentContext] CGContext])
         return;
-
     m_savedNSGraphicsContext = [NSGraphicsContext currentContext];
-    NSGraphicsContext* newContext = [NSGraphicsContext graphicsContextWithCGContext:cgContext flipped:isFlipped];
-    [NSGraphicsContext setCurrentContext:newContext];
+    RetainPtr<NSGraphicsContext> newContext { ([NSGraphicsContext graphicsContextWithCGContext:cgContext flipped:isFlipped]) };
```

Comment:
Why is `RetainPtr` necessary here?

Reply:
Just to clarify the ownership. Mentioned in the commit message..

Reply:
I'm not sure we really prefer additional ref-churn over simply using autorelease? Because this is still going to autorelease.